runtime/near-test-contracts/build.rs

```diff
@@ -48,3 +44,16 @@ fn cargo_build_cmd() -> Command {
     res.args(&["build", "--target=wasm32-unknown-unknown", "--release"]);
     res
 }
+
+fn check_status(mut cmd: Command) -> io::Result<()> {
+    let status = cmd.status().map_err(|err| {
+        io::Error::new(io::ErrorKind::Other, format!("command `{:?}` failed to run: {}", cmd, err))
+    })?;
+    if !status.success() {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            format!("command `{:?}` exited with non-zero status: {:?}", cmd, status),
```

Comment:
Directly output vector looks ugly, and user cannot copy and paste to shell to rerun & debug. Use cmd.join(" ") or use a more correct `shlex.join`: https://docs.rs/shlex/1.0.0/shlex/fn.join.html?

Reply:
Agree that it's a bit ugly, but it is ugly specifically such that copy&paste works even if there are things spaces in the command name. 

```
λ git diff
diff --git a/runtime/near-test-contracts/build.rs b/runtime/near-test-contracts/build.rs
index 116c61cfb..78f966b36 100644
--- a/runtime/near-test-contracts/build.rs
+++ b/runtime/near-test-contracts/build.rs
@@ -10,7 +10,7 @@ fn main() {
 
 fn try_main() -> io::Result<()> {
     let mut cmd = Command::new("rustup");
-    cmd.args(&["target", "add", "wasm32-unknown-unknown"]);
+    cmd.args(&["target", "addd", "wasm32-unknown-unknown"]);
     check_status(cmd)?;
 
     build_contract("./test-contract-rs", &[], "test_contract_rs")?;

20:13:46|~/projects/nearcore|better-build-errors⚡*
λ cargo build -q -p near-test-contracts
error: failed to run custom build command for `near-test-contracts v0.0.0 (/home/matklad/projects/nearcore/runtime/near-test-contracts)`

Caused by:
  process didn't exit successfully: `/home/matklad/projects/nearcore/target/debug/build/near-test-contracts-f8128a735147dfcb/build-script-build` (exit code: 1)
  --- stdout
  error: The subcommand 'addd' wasn't recognized
  	Did you mean 'add'?

  If you believe you received this message in error, try re-running with 'rustup target -- addd'

  USAGE:
      rustup target <SUBCOMMAND>

  For more information try --help

  --- stderr
  command `"rustup" "target" "addd" "wasm32-unknown-unknown"` exited with non-zero status: ExitStatus(ExitStatus(256))

20:13:48|~/projects/nearcore|better-build-errors⚡*
λ "rustup" "target" "addd" "wasm32-unknown-unknown"
error: The subcommand 'addd' wasn't recognized
	Did you mean 'add'?

If you believe you received this message in error, try re-running with 'rustup target -- addd'

USAGE:
    rustup target <SUBCOMMAND>

For more information try --help
```

I don't think it's worth pulling an extra dependency here. JFYI, a more general solution for "I want to run a bunch of shell commands in rust and get reasonable errors bace" would be https://github.com/matklad/xshell


A proper solution here would be to implement `Display` for `std::process::Command`, thats on my todo list: https://github.com/rust-lang/rust/issues/72009.